include/revng/Storage/Path.h

```diff
@@ -0,0 +1,142 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/SmallString.h"
+#include "llvm/Support/Path.h"
+
+#include "revng/Storage/StorageClient.h"
+#include "revng/Support/Debug.h"
+#include "revng/Support/PathList.h"
+
+namespace revng {
+
+class BasePath {
```

Comment:
I've added the `Base` to stress that it's the parent of `FilePath` and `DirectoryPath` and to avoid confusion like "Ah it's a `Path`, I can use it to do file operations" whereas `BasePath` is more explicit

Reply:
Right, I though it was a abstract class, but it's not. Change to `PathBase`.

Reply:
Renamed